taigapy/client_v3.py

```diff
@@ -314,6 +314,31 @@ def _get(self, datafile_id: str) -> pd.DataFrame:
             )
 
         return result
+    
+    def upload_to_gcs(self, data_file_taiga_id: str, dest_gcs_path: str) -> bool:
+        """Upload a Taiga datafile to a specified location in Google Cloud Storage.
+
+        The service account taiga-892@cds-logging.iam.gserviceaccount.com must have
+        storage.buckets.create access for this request.
+
+        Arguments:
+            `data_file_taiga_id` -- Taiga ID in the form dataset_permaname.dataset_version/datafile_name
+            `dest_gcs_path` -- Google Storage path to upload to, in the form bucket:path
+
+        Returns:
+            bool -- Whether the file was successfully uploaded
+        """
+        full_taiga_id = self.get_canonical_id(data_file_taiga_id)
+        if full_taiga_id is None:
+            return False
+        
+        try:
+            self.api.upload_to_gcs(full_taiga_id, dest_gcs_path)
```

Comment:
This looks like probably a copy the corresponding function in the V2 of the client, correct?

If we call this, we're going to upload the file in the format that it is internally stored in Taiga. In the case of our releases, that's not what we want. (ie: some files are now stored in HDF5 but when we upload to GCS we want to upload a CSV version of the file)

This is why I said we should make a call to download_to_cache and then do the upload ourselves instead of using this API call which asks the Taiga service to do the upload.

Reply:
Thinking about this a little more, I think this also means we'll need to add a `requested_format` onto this function as well so that we don't assume what format that we want uploaded to GCS.

Reply:
@pgm I've updated the function